diff --git a/docker/container.go b/docker/container.go
index 95b4e2b..794ef28 100644
--- a/docker/container.go
+++ b/docker/container.go
@@ -387,7 +387,7 @@ func (c *Container) Log() error {
 func (c *Container) pull(image string) error {
 	taglessRemote, tag := parsers.ParseRepositoryTag(image)
 	if tag == "" {
-		image = utils.ImageReference(taglessRemote, tags.DEFAULTTAG)
+		image = utils.ImageReference(taglessRemote, tags.DefaultTag)
 	}
 
 	repoInfo, err := registry.ParseRepositoryInfo(taglessRemote)
diff --git a/docker/convert.go b/docker/convert.go
index 82aa04f..39e0c86 100644
--- a/docker/convert.go
+++ b/docker/convert.go
@@ -9,6 +9,7 @@ import (
 	"github.com/docker/libcompose/project"
 	"github.com/docker/libcompose/utils"
 	"github.com/samalba/dockerclient"
+	"github.com/docker/docker/pkg/stringutils"
 )
 
 func Filter(vs []string, f func(string) bool) []string {
@@ -80,25 +81,25 @@ func Convert(c *project.ServiceConfig) (*runconfig.Config, *runconfig.HostConfig
 	}
 
 	config := &runconfig.Config{
-		Entrypoint:   runconfig.NewEntrypoint(c.Entrypoint.Slice()...),
+		Entrypoint:   stringutils.NewStrSlice(c.Entrypoint.Slice()...),
 		Hostname:     c.Hostname,
 		Domainname:   c.DomainName,
 		User:         c.User,
 		Env:          c.Environment.Slice(),
-		Cmd:          runconfig.NewCommand(c.Command.Slice()...),
+		Cmd:          stringutils.NewStrSlice(c.Command.Slice()...),
 		Image:        c.Image,
 		Labels:       c.Labels.MapParts(),
 		ExposedPorts: ports,
 		Tty:          c.Tty,
 		OpenStdin:    c.StdinOpen,
 		WorkingDir:   c.WorkingDir,
-		VolumeDriver: c.VolumeDriver,
+		// TODO: VolumeDriver: c.VolumeDriver,
 		Volumes:      volumes,
 	}
 	host_config := &runconfig.HostConfig{
 		VolumesFrom: c.VolumesFrom,
-		CapAdd:      runconfig.NewCapList(c.CapAdd),
-		CapDrop:     runconfig.NewCapList(c.CapDrop),
+		CapAdd:      stringutils.NewStrSlice(c.CapAdd...),
+		CapDrop:     stringutils.NewStrSlice(c.CapDrop...),
 		CPUShares:   c.CpuShares,
 		CpusetCpus:  c.CpuSet,
 		ExtraHosts:  c.ExtraHosts,
