diff --git a/docker/builder.go b/docker/builder.go
index 7bd8d13..f4b0e40 100644
--- a/docker/builder.go
+++ b/docker/builder.go
@@ -136,7 +136,8 @@ func CreateTar(p *project.Project, name string) (io.ReadCloser, error) {
 	}
 	var includes = []string{"."}
 
-	excludes, err := utils.ReadDockerIgnore(path.Join(root, ".dockerignore"))
+	dockerignore, err := os.Open(path.Join(root, ".dockerignore"))
+	excludes, err := utils.ReadDockerIgnore(dockerignore)
 	if err != nil {
 		return nil, err
 	}
diff --git a/docker/convert.go b/docker/convert.go
index b5fb765..7c69006 100644
--- a/docker/convert.go
+++ b/docker/convert.go
@@ -5,6 +5,7 @@ import (
 
 	"github.com/Sirupsen/logrus"
 	"github.com/docker/docker/pkg/nat"
+	"github.com/docker/docker/pkg/stringutils"
 	"github.com/docker/docker/runconfig"
 	"github.com/docker/libcompose/project"
 	"github.com/docker/libcompose/utils"
@@ -84,25 +85,25 @@ func Convert(c *project.ServiceConfig) (*runconfig.Config, *runconfig.HostConfig
 	}
 
 	config := &runconfig.Config{
-		Entrypoint:   runconfig.NewEntrypoint(c.Entrypoint.Slice()...),
+		Entrypoint:   stringutils.NewStrSlice(c.Entrypoint.Slice()...),
 		Hostname:     c.Hostname,
 		Domainname:   c.DomainName,
 		User:         c.User,
 		Env:          c.Environment.Slice(),
-		Cmd:          runconfig.NewCommand(c.Command.Slice()...),
+		Cmd:          stringutils.NewStrSlice(c.Command.Slice()...),
 		Image:        c.Image,
 		Labels:       c.Labels.MapParts(),
 		ExposedPorts: ports,
 		Tty:          c.Tty,
 		OpenStdin:    c.StdinOpen,
 		WorkingDir:   c.WorkingDir,
-		VolumeDriver: c.VolumeDriver,
-		Volumes:      volumes,
+		// TODO: VolumeDriver: c.VolumeDriver,
+		Volumes: volumes,
 	}
 	hostConfig := &runconfig.HostConfig{
 		VolumesFrom: c.VolumesFrom,
-		CapAdd:      runconfig.NewCapList(c.CapAdd),
-		CapDrop:     runconfig.NewCapList(c.CapDrop),
+		CapAdd:      stringutils.NewStrSlice(c.CapAdd...),
+		CapDrop:     stringutils.NewStrSlice(c.CapDrop...),
 		CPUShares:   c.CPUShares,
 		CpusetCpus:  c.CPUSet,
 		ExtraHosts:  c.ExtraHosts,
